packages/unimodules-test-core/android/src/main/java/org/unimodules/test/core/TestUtils.kt

```diff
@@ -0,0 +1,46 @@
+package org.unimodules.test.core
+
+import android.os.Bundle
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertTrue
+import junit.framework.ComparisonFailure
+import org.unimodules.core.arguments.MapArguments
+import org.unimodules.core.arguments.ReadableArguments
+
+fun assertSetsEqual(first: Set<*>, second: Set<*>, message: String = "") {
+  if (!first.all { second.contains(it) }) {
+    throw ComparisonFailure(message, first.toString(), second.toString())
+  }
+}
+
+fun assertListsEqual(first: List<*>?, second: List<*>?, message: String = "") {
+  if (first == second) return
+
+  if (first == null || second == null) {
+    throw throw ComparisonFailure(message, first.toString(), second.toString())
+  }
+
+  if (!first.toTypedArray().contentDeepEquals(second.toTypedArray())) {
+    throw ComparisonFailure(message, first.toString(), second.toString())
+  }
+}
+
+fun promiseResolved(promise: PromiseMock, with: (Bundle) -> Unit) {
+  assertEquals(PromiseState.RESOLVED, promise.state)
+  with(promise.resolveValue as Bundle)
```

Comment:
I'm not sure if we want to shadow some std lib functionality here: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/with.html? I dislike it to some extends 😉 On the other hand it's the most correct word here 😋

Reply:
I wouldn't worry too much, since it is very limited scope. I can change it if you'll insist ;)